### PR TITLE
Remove unused code and methods' improve visibility in MinerServerImpl

### PR DESCRIPTION
**Pull request by tinchou**
It was intended to be part of the `fallback mining` feature but never completed. This removed the unused code.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 4 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L342](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L342): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L346](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L346): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L137](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L137): Constructor has 12 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L315](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L315): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L316](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L316): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L341](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L341): Incorrect lazy initialization of static field co.rsk.mine.MinerServerImpl.privKey0 in co.rsk.mine.MinerServerImpl.generateFallbackBlock() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ALI_LAZY_INIT_STATIC)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L342](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L342): java/io/File.<init>(Ljava/lang/String;Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L345](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L345): Incorrect lazy initialization of static field co.rsk.mine.MinerServerImpl.privKey1 in co.rsk.mine.MinerServerImpl.generateFallbackBlock() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ALI_LAZY_INIT_STATIC)
1. ![MAJOR][MAJOR] [MinerServerImpl.java#L346](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L346): java/io/File.<init>(Ljava/lang/String;Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [MinerUtils.java#L73](https://github.com/rsksmart/rskj/blob/18c26fda0bd8a68af9d0351a7917d1b8ba77bf01/rskj-core/src/main/java/co/rsk/mine/MinerUtils.java#L73): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java [hunk 1]
```diff
@@ -107,55 +106,6 @@ public void buildBlockToMineCheckThatLastTransactionIsForREMASC() {
         assertThat(remascTransaction, instanceOf(RemascTransaction.class));
     }
 
-
-
-    @Test
-    public void submitBitcoinBlockTwoTags() {
```
Comment (SergioDemianLerner) [inline rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java]: Este test lo agregue yo y me parecio importantisimo porque prueba un potencial problema d e seguridad que podria ocurrir si hay mas de un tag RSK

File: rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java [hunk 2]
```diff
@@ -346,13 +296,6 @@ public void increaseTimeTwice() {
         return MinerUtils.getBitcoinMergedMiningBlock(bitcoinNetworkParameters, bitcoinMergedMiningCoinbaseTransaction);
     }
 
-    private co.rsk.bitcoinj.core.BtcBlock getMergedMiningBlockWithTwoTags(MinerWork work,MinerWork work2) {
```
Comment (SergioDemianLerner) [inline rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java]: parte necesaria del test

File: rskj-core/src/main/java/co/rsk/mine/MinerUtils.java [hunk 3]
```diff
@@ -77,56 +77,6 @@
         return coinbaseTransaction;
     }
 
-    public static co.rsk.bitcoinj.core.BtcTransaction getBitcoinMergedMiningCoinbaseTransactionWithTwoTags(
```
Comment (SergioDemianLerner) [inline rskj-core/src/main/java/co/rsk/mine/MinerUtils.java]: Part of a test case. No need to take out.

**Comment by tinchou** [comment]
@SergioDemianLerner it's hurting us because it's pretty hard to make any changes to these methods. I'm trying to extract the block generation logic but everything's tangled.

E.g.: `buildBlockToMine` calls `setFallbackMiningState`, which calls `buildBlockToMine` again... and then what? it's hard to follow.

I removed the whole test because it seemed to rely on the `extraData` thing, which wasn't used anywhere in production. It also relied on `getBitcoinMergedMiningCoinbaseTransactionWithTwoTags`, which for some reason was left in **Production code** when it is not used in production.

Maybe you can help me separate the part of the test that is important for security purposes?

**Comment by SergioDemianLerner** [comment]
The getBitcoinMergedMiningCoinbaseTransactionWithTwoTags() is in MinerUtils. You can move it to some file named TestMinerUtils, and so it doesn't exists in production code.

**Comment by tinchou** [comment]
I'd like to do that! But how do I change the test to work without the `extraData` flag?

**Comment by SergioDemianLerner** [comment]
Does it check anything related to that flag? I supposed that it didn't. It just set it, but not check it.

**Comment by tinchou** [comment]
@SergioDemianLerner is it OK this way? I just removed the check that the two blocks should be equal: https://github.com/rsksmart/rskj/pull/477/commits/cb7e23653327a090853359068365e9cc93ffaff2

**Comment by tinchou** [comment]
Closing because this isn't a priority for our current tasks, but should definitely be done at some point.
